BXMSDOC-4618-master: Update process doc content for Michele for 7.5.

sterobin opened the pull request:
Links:
* [Jira](https://issues.jboss.org/browse/BXMSDOC-4618) - And [SME JIRA](https://issues.jboss.org/browse/BXMSDOC-4634) and [QE JIRA](https://issues.jboss.org/browse/BXMSDOC-4635)
* [PAM doc preview](http://file.rdu.redhat.com/~sterobin/BXMSDOC-4618_PAM/)
* [jBPM doc preview](http://file.rdu.redhat.com/~sterobin/BXMSDOC-4618_JBPM/#business_process)  -  Same, but for community

@krisv @LuboTerifaj Tagged you for your overall review on Michele's doc. **You can skip all of chapters 2 and 3 on BPMN2 ref content.** That is single-sourced from the BPMN ref doc and is not new. I've clarified things, fixed some issues, peer reviewed, and added the new section on "Invoking a DMN service in a business process", which I intended to add soon anyway from the Drools side, compliments of @jomarko.

@jomarko, can you review and verify the [Invoking a DMN service in a business process](http://file.rdu.redhat.com/~sterobin/BXMSDOC-4618_PAM/#dmn-execution-business-process) section? I decided to roll with your same example for simplicity, just clarified some things and made consistent.

jomarko commented:
@sterobin thank you. It looks good to me. My only point, as you can see in my notes https://issues.jboss.org/browse/BXMSDOC-4558, is about referring **Route To Rail** node. You refer as **DMN service node** but not sure if it is an issue. Leaving the decision up to you.

sterobin commented:
@jomarko, I pushed the change we agreed on in chat, and updated the preview. Thanks you!

sterobin commented:
Thanks @LuboTerifaj! I've implemented your feedback, updated the doc previews, and replied to you in the [JIRA](https://issues.jboss.org/browse/BXMSDOC-4635). Let me know what you think.

sterobin commented:
Received @lazarotti's approval in JIRA and chat.
